### DISCUSSION TITLE
Runnning on Nano 33 IoT board

### DISCUSSION THREAD

**Discussion by MigraineMan33** [marcel-licence/samd21_mini_synth#3] [Show and tell]
Cloned the min-synth project and targeted a Nano 33 IoT ... figured it has the same SAMD21 processor and should port pretty easily.  I installed a clean version of Arduino 1.8.16, and added the SAMD library support.  First compile resulted in a bunch of errors about "Serial.printf doesn't exist."  Not sure if there's a cleaner way to resolve this, but message-only print statements were changed to Serial.print, and the prints with parameters were modified as shown below -
>inline void Midi_PitchBend(uint8_t ch, uint16_t bend)
> {
>     **char buffer [64]; //** 
>     float value = ((float)bend - 8192.0f) * (1.0f / 8192.0f) - 1.0f;
>     if (midiMapping.pitchBend != NULL)
>     {
> **//        Serial.printf("PB: %d, %d, %d\n", ch, bend, (value * 1000.0f));
>         sprintf(buffer, "PB: %d, %d, %d\n", ch, bend, (value * 1000.0f));
>         Serial.print(buffer);**
>         midiMapping.pitchBend(ch, value);
>     }
> }

I changed the serial baud rate in "midi_interface" to be 31250 instead of 115200, because that's what my cheesy USB->MIDI converter was putting out.  I also fixed two issues at the top of "samd21_mini_synth" in the setup section:
Serial.begin should have 115200 as the parameter, not 11500
The built-in LED pin initialization is missing - there's no 1Hz blinkage.  Add:
pinMode(LED_BUILTIN, OUTPUT); // needed to blink 1Hz
![20211126_150829](https://user-images.githubusercontent.com/95103775/143627408-3a1dae73-822f-4de5-aae6-38752237f3fc.jpg)
![20211126_150851](https://user-images.githubusercontent.com/95103775/143627419-56d05a34-d826-477f-bd1e-e578774dd5a7.jpg)
![20211126_150910](https://user-images.githubusercontent.com/95103775/143627433-00748049-3941-48e9-871e-f08f232ba8ea.jpg)

**Comment by marcel-licence** [comment]
Thank you very much sharing your investigation. Good to hear that it works.
I've added the missing line locally `pinMode(LED_BUILTIN, OUTPUT); // needed to blink 1Hz` and will commit this next time.
Hm I am not sure if the XIAO just ignores the baudrate, because I didn't reconize my mistake. 

Thanks

**Reply by diyelectromusic** [reply]
I've been trying to run it on a XIAO and mine also defaults to 115200 for serial MIDI, so I had to change it to 31250.  I added it as an extra setting in config.h, but I'd have thought the default in mini_interface probably should be 31250 unless overridden by config.h as that is the MIDI spec value of course.

Kevin

**Comment by MigraineMan33** [comment]
Marcel,
Just received an ESP32 Audio Kit board, and was successful getting the code
to run after sleuthing out which of the codec implementations I have.
Hoping to get some time to work on this project over the holidays.  And I
want to send you a big "thank you" for helping me figure out how to run
multithreaded on the ESP32 - your project provided an easy to follow
example of multi-core, multithreaded implementation.

Best regards and Merry Christmas,
MM

**Reply by marcel-licence** [reply]
Hi MM,

you are welcome! I am very happy about your feedback.

Wish you also a peaceful time and a Merry Christmas
